version_tag_check/version_tag_check_action.py

```diff
@@ -62,14 +63,23 @@ def run(self) -> None:
         existing_versions: list[Version] = repository.get_all_tags()
 
         if new_version in existing_versions:
-            logger.error("The tag already exists in repository.")
-            sys.exit(1)
+            if not self.should_exist:
+                logger.error("The tag already exists in the repository.")
+                sys.exit(1)
+        else:
+            if self.should_exist:
+                logger.error("The tag does not exist in the repository.")
+                sys.exit(1)
+
+        if self.should_exist:
+            sys.exit(0)
```

Comment:
This whole block of IF statements is weird. It took me quite some time to understand. Cause it is not well readable,  please add necessary comments. In my opinion `if not self.should_exist:` doesn't have to be under `if new_version in existing_versions:` since it doesn't have else statement. Everytime self.should_exist is False, than we quit the run. And than we can check `if new_version NOT in existing_versions:`. What do you think?

Reply:
Fixed in commit ab6cd3a3d53c8332c19bc1ffb109bf862f5b53ff.